### ISSUE TITLE
cfn-lint doesn't validate second level key name of a FindInMap that is resolved by a nested FindInMap

### ISSUE THREAD

**Issue by Slooz** [aws-cloudformation/cfn-lint#2078]
*cfn-lint version: 0.52.0*

cfn-lint doesn't catch the following error:
```yaml
Mappings:
  Mapping:
    Key:
      BucketNameKeyName: "" # should be BucketName
      BucketName: Bucket
Resources:
  S3Bucket:
    Type: AWS::S3::Bucket
    Properties:
      BucketName: !FindInMap
        - Mapping
        - Key
        - !FindInMap
          - Mapping
          - Key
          - BucketNameKeyName
```

But the [`validate-template` API](https://awscli.amazonaws.com/v2/documentation/api/latest/reference/cloudformation/validate-template.html) does:
> Template error: Unable to get mapping for Mapping::Key::

I was surprised that the template passed cfn-lint but failed `validate-template`. Is this something that cfn-lint could validate?

We use this syntax so that we can do something like the following, which is to have a mapping with an extra dimension based on deployment names within an account:
```yaml
Parameters:
  DeploymentName:
    Type: String
Mappings:
  AccountMap:
    "000000000001":
      PrimaryDeploymentBucketName: Account1PrimaryBucket
      SecondaryDeploymentBucketName: Account1SecondaryBucket
      TertiaryDeploymentBucketName: Account1TertiaryBucket
    "000000000002":
      PrimaryDeploymentBucketName: Account2PrimaryBucket
      SecondaryDeploymentBucketName: Account2SecondaryBucket
      TertiaryDeploymentBucketName: Account2TertiaryBucket
  DeploymentMap:
    Primary:
      BucketNameAccountMapKey: PrimaryDeploymentBucketName
    Secondary:
      BucketNameAccountMapKey: SecondaryDeploymentBucketName
    Tertiary:
      BucketNameAccountMapKey: TertiaryDeploymentBucketName
Resources:
  S3Bucket:
    Type: AWS::S3::Bucket
    Properties:
      BucketName: !FindInMap
        - AccountMap
        - !Ref AWS::AccountId
        - !FindInMap
          - DeploymentMap
          - !Ref DeploymentName
          - BucketNameAccountMapKey
```

**Comment by Slooz** [comment]
I don't think the issue here is that cfn-lint can't find `AWS::AccountId` in the `AccountMap`. The second code example that uses `AWS::AccountId` is just an example of how we use this functionality.

The first code example shows that `cfn-lint` doesn't catch the error, but `validate-template` does.

However, I do agree with you, it'd be useful if a list of account IDs could be provided similarly to how a list of regions can be provided.
